src/api/spec/controllers/webui/comments_controller_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'rails_helper'
+
+RSpec.describe Webui::CommentsController, type: :controller do
+  let(:user) { create(:confirmed_user, login: 'luck') }
+
+  describe 'POST #create' do
+    let!(:project) { create(:project, name: 'openSUSE') }
+    let!(:package) { create(:package, name: 'osc', project: project) }
+    let!(:bs_request) { create(:bs_request, description: "Please take this", creator: user.login) }
+
+    before do
+      login user
+    end
+
+    %w(project package bs_request).each do |klass|
+      context "saving a comment of a #{klass}" do
+        let(:commentable) { klass.classify.constantize.first }
+
+        before do
+          params = { comment: { body: "This #{klass} is AWESOME!" }, commentable_type: commentable.class, commentable_id: commentable.id }
+          post :create, params: params
+        end
+
+        it 'should be created' do
+          expect(flash[:notice]).to eq('Comment was successfully created.')
+          expect(commentable.comments.first.body).to eq("This #{klass} is AWESOME!")
+        end
+      end
+    end
+
+    context 'saving a comment without body' do
+      before do
+        params = { comment: { body: '' }, commentable_type: package.class, commentable_id: package.id }
+        post :create, params: params
+      end
+
+      it 'should not be created' do
+        expect(flash[:error]).to eq('Comment can\'t be saved: Body can\'t be blank.')
+        expect(package.comments.count).to eq(0)
+      end
+    end
+
+    context 'saving a comment of a non-commentable class' do
+      let(:distribution) { create(:distribution) }
+
+      it 'does not allow to create a comment' do
+        params = { comment: { body: 'Wonderful Package' }, commentable_type: distribution.class, commentable_id: distribution.id }
+        expect{ post :create, params: params }.to raise_error(NoMethodError)
+      end
+    end
+
+    it 'does not allow to overwrite the comment\'s user' do
+      params ={ comment: { body: 'This project is AWESOME!', user_id: user }, commentable_type: project.class, commentable_id: project.id }
+      expect { post :create, params: params }.to raise_error(ActionController::UnpermittedParameters)
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    let(:admin) { create(:admin_user, login: 'Admin') }
+    let(:user_tom) { create(:confirmed_user, login: 'tom') }
```

Comment:
You don't need ``user_tom`` as you only use it to create ``tom_comment``, however, the ``comment_project`` factory will create a user by default. Please don't forget to rename ``tom_comment`` then!